### DISCUSSION TITLE
recursive search in json

### DISCUSSION THREAD

**Discussion by PFlorient** [TarekRaafat/autoComplete.js#263] [Q&A]
Hello everyone,

I have a little problem with an api. In fact this one returns me several different names. A name "parent" and a name "child" in this way.  

```
[
   {
      "id":"1",
      "name":"Room 50",
      "sites":[
         {
            "id":"1",
            "name":"Guillonboeuf"
         },
         {
            "id":"2,
            "name":"Joly-sur-Bertin"
         }
      ]
   },
   {
      "id":"2",
      "name":"Room 55",
      "sites":[
         {
            "id":"1",
            "name":"Bernier-sur-Mer"
         },
         {
            "id":"2",
            "name":"Grenier"
         }
      ]
   },
   {
      "id":"3",
      "name":"Room 52",
      "sites":[
         {
            "id":"1",
            "name":"Truc"
         },
         {
            "id":"2",
            "name":"Joubert-sur-Mer"
         }
      ]
   }
]
```

But I can only find results with the parent name.

Example: I find Room 50... If I write "R". But no result if I write "Guillonboeuf". 

Did I miss something, how can i find something in a json array  ?

Thank you in advance for your answer !

**Comment by folknor** [comment]
autoComplete.js searches only the first level of the array using entries in the data returned from `config.data` as defined in `config.keys`. If you need to search deeper, you must do so manually before you return the data from `config.data`, or change the REST service backend.

But this could be changed, of course. autoComplete.js could change the `config.keys` syntax to accept trees, like;
```js
new autoComplete({
	data: {
		keys: [
			"name",
			["sites", "name"]
		]
	}
})
```

**Reply by PFlorient** [reply]
Thank you for responding so quickly.

The tree search seems to be more coherent haha.

However, an error occurred as a result of this operation, when I start writing I have this :
`Uncaught (in promise) TypeError: can't access property "toString", t is undefined`

Do you think this one is related? Or have I overlooked something else?

**Reply by folknor** [reply]
I do not understand your comment.

What do you mean by "as a result of this operation"?

Can you post some of your code?

I'm also worried that you misunderstood my last comment. Please read it again.

**Reply by PFlorient** [reply]
Sorry, I am not very fluent in english, so yes I did miss something...

```JavaScript
new autoComplete({
                data: {
                    selector: "#input-search-dataroom",
                    src: async () => {
                        try {
                          document
                            .getElementById("input-search-dataroom")
                            .setAttribute("placeholder", "Recherche");
                          const source = await fetch(
                            'localhost:80/api/dataroom/'
                          );
                          const data = await source.json();
                          return data;
                        } catch (error) {
                          return error;
                        }
                      },
                    keys: [
                        "name",
                        ["sites", "name"]
                    ],
                    cache: true,
                },
              ...
});
```

**Reply by folknor** [reply]
Yes, autoComplete.js does not work like this now.

I said @TarekRaafat (who makes autoComplete.js) could maybe change it in the future, so that this can work in the next version (autoComplete.js version 10.3 or 10.2.7, next release):
```js
		keys: [
			"name",
			["sites", "name"]
		]
```

It does not work now in 10.2.6.

Do you understand? English is not my language either :-)
